src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx

```diff
@@ -0,0 +1,380 @@
+import React from 'react';
+import { mount, configure } from 'enzyme';
+import { Input } from 'antd';
+import lodash from 'lodash';
+import * as sdk from '@cognite/sdk';
+import TimeserieSearchAndSelect from './TimeseriesSearchAndSelect';
+import DetailCheckbox from 'components/DetailCheckbox/DetailCheckbox';
+import { assetsList } from 'mocks/assetsList';
+import { timeseriesList } from 'mocks/timeseriesList';
+import Adapter from 'enzyme-adapter-react-16';
+import Mock = jest.Mock;
+
+configure({ adapter: new Adapter() });
+
+sdk.TimeSeries.search = jest.fn();
+sdk.Assets.list = jest.fn();
+
+const { Search } = Input;
+const propsCallbacks: { [name: string]: Mock } = {
+  filterRule: jest.fn(),
+  onTimeserieSelectionChange: jest.fn(),
+  onError: jest.fn(),
+};
+
+// ignore debounce
+jest.spyOn(lodash, 'debounce').mockImplementation((f: any) => {
+  return f;
+});
+
+beforeEach(() => {
+  // @ts-ignore
+  sdk.TimeSeries.search.mockResolvedValue({ items: timeseriesList });
+  // @ts-ignore
+  sdk.Assets.list.mockResolvedValue({ items: assetsList });
+});
+
+afterEach(() => {
+  Object.keys(propsCallbacks).forEach((key: string) =>
+    propsCallbacks[key].mockClear()
+  );
+  // @ts-ignore
+  sdk.TimeSeries.search.mockClear();
+});
+
+
+describe('TimeserieSearchAndSelect', () => { // tslint:disable-line
```

Comment:
Why `tslint:disable-line`?

Reply:
tslint complains that this function is too many lines, and it should be broken up into more functions. Generally I think that is a good idea, but limiting a describe block in tests seems wrong. Would be better to disable that exact rule, but I haven't been able to find out what the rule is called

Reply:
Found the rule name. Will update PR

Reply:
Done